src/Compilers/CSharp/Portable/Symbols/Source/SourceClonedParameterSymbol.cs

```diff
@@ -10,6 +10,9 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.Symbols
 {
+
+    // PROTOTYPE(caller-expr): Make SourceClonedParameterSymbol abstract and introduce new specialized types (same as being done in VB).
```

Comment:
I'll do this once the corresponding VB change is reviewed.

Reply:
@AlekseyTs Let's keep the remaining C# work in a separate PR?

Reply:
➡️ https://github.com/dotnet/roslyn/pull/54355 (let me know if you want everything here).

Reply:
> (let me know if you want everything here).

I have no objections aganst separation of changes